packages/headless/src/modules/user/repository/user.db.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@nestjs/common';
+import { UserEntity } from 'src/entity/user';
+
+@Injectable()
+export abstract class IUserDB {
+  abstract save: (body: UserEntity) => Promise<UserEntity>;
+  abstract findOne: (query: Record<string, string>) => Promise<UserEntity>;
+  abstract findOneForLogin: (
```

Comment:
findOne & findOneForLogin are same logic.So why do you use twice them?...
And you may use a proper name like findUser instead of those functions

Reply:
findOneForLogin returns password others don't.